src/browser/web-progress.js

```diff
@@ -0,0 +1,122 @@
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define((require, exports, module) => {
+
+  'use strict';
+
+  const {Record, Union} = require('common/typed');
+  const {StyleSheet, Style} = require('common/style');
+  const {html} = require('reflex');
+
+  // Model
+  const Model = Record({
+    loadStarted: false,
+    loadEnded: false,
+  });
+  exports.Model = Model;
+
+  // Action
+
+  const LoadStart = Record({
+    id: String,
+  }, 'WebView.WebProgress.LoadStart');
+
+  const LoadEnd = Record({
+    id: String,
+  }, 'WebView.WebProgress.LoadEnd');
+
+  const Action = Union({LoadStart, LoadEnd});
+  exports.Action = Action;
+
+  const update = (state, action) =>
+    action instanceof LoadStart ? state.merge({
+      loadStarted: true,
+      loadEnded: false
+      }) :
```

Comment:
Nit: Can we please move **then** block to the next line when it does not fit on the same line to match code elsewhere, so this would become:

```js
action instanceof LoadStart ?
  state.merge({loadStarted: true, loadEnded: false}) :
action instanceof LoadEnd ?
  state.merge({loadEnded: true})
state;
```